### PR TITLE
add SHARD_ROW_ID_BITS

### PR DESCRIPTION
**Pull request by kissmydb**
SHARD_ROW_ID_BITS is lost in v4.0

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### CODE REVIEW

File: shard_row_id_bits.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+---
+title: SHARD_ROW_ID_BITS
+category: reference
```
Comment (TomShawn) [inline shard_row_id_bits.md:3]: summary 应该在 category 上面

File: shard_row_id_bits.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+summary: 介绍 TiDB 的 `SHARD_ROW_ID_BITS` 表属性。
+---
+
+# SHARD_ROW_ID_BITS
+
+本文介绍表属性 `SHARD_ROW_ID_BITS` ，它用来设置隐式 _tidb_rowid 的分片数量的 bit 位数
```
Comment (TomShawn) [inline shard_row_id_bits.md:9]:
```suggestion
本文介绍表属性 `SHARD_ROW_ID_BITS`，它用来设置隐式 `_tidb_rowid` 分片数量的 bit 位数。
```

File: shard_row_id_bits.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+## 基本概念
+
+对于 PK 非整数或没有 PK 的表，TiDB 会使用一个隐式的自增 rowid，大量 `INSERT` 时会把数据集中写入单个 Region，造成写入热点。
```
Comment (TomShawn) [inline shard_row_id_bits.md:13]: PK 是否有全称？
```suggestion
对于 PK 非整数或没有 PK 的表，TiDB 会使用一个隐式的自增 rowid。大量执行 `INSERT` 插入语句时会把数据集中写入单个 Region，造成写入热点。
```

File: shard_row_id_bits.md [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+通过设置 `SHARD_ROW_ID_BITS`，可以把 rowid 打散写入多个不同的 Region，缓解写入热点问题。但是设置的过大会造成 RPC 请求数放大，增加 CPU 和网络开销。
+
+- `SHARD_ROW_ID_BITS = 4` 表示 16 个分片
+- `SHARD_ROW_ID_BITS = 6` 表示 64 个分片
+- `SHARD_ROW_ID_BITS = 0` 表示默认值 1 个分片
+
+## 语句示例
+
+- `CREATE TABLE`：`CREATE TABLE t (c int) SHARD_ROW_ID_BITS = 4;`
```
Comment (TomShawn) [inline shard_row_id_bits.md:23]: 写成

```sql
CREATE TABLE t (c int) SHARD_ROW_ID_BITS = 4;
```

会不会更好一些？

**Comment by jackysp** [comment]
Please address comments @kissmydb

File: shard_row_id_bits.md [hunk 5]
```diff
@@ -0,0 +1,24 @@
+---
```
Comment (yikeke) [inline shard_row_id_bits.md:1]: Please:
1. rename this file to shard-row-id-bits.md.
2. Add this file to TOC.md. 

@kissmydb

File: shard_row_id_bits.md (hunk 4 above)
Comment (kissmydb) [reply shard_row_id_bits.md:23]: 先保持原样好一些，表示了一个分类

File: shard-row-id-bits.md [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+category: reference
+---
+
+# SHARD_ROW_ID_BITS
+
+本文介绍表属性 `SHARD_ROW_ID_BITS`，它用来设置隐式 `_tidb_rowid` 分片数量的 bit 位数。
+
+## 基本概念
+
+对于 PK 非整数或没有 PK 的表，TiDB 会使用一个隐式的自增 rowid。大量执行 `INSERT` 插入语句时会把数据集中写入单个 Region，造成写入热点。
```
Comment (yikeke) [inline shard-row-id-bits.md:13]:
```suggestion
对于主键非整数或没有主键的表，TiDB 会使用一个隐式的自增 rowid。大量执行 `INSERT` 插入语句时会把数据集中写入单个 Region，造成写入热点。
```

File: faq/sql-faq.md [hunk 7]
```diff
@@ -214,7 +214,7 @@ ID 没什么规律，只要是唯一就行，不过生成的时候，是有一
 
 ### 如何打散热点
 
-TiDB 中以 Region 分片来管理数据库，通常来讲，TiDB 的热点指的是 Region 的读写访问热点。而 TiDB 中对于 PK 非整数或没有 PK 的表，可以通过设置 `SHARD_ROW_ID_BITS` 来适度分解 Region 分片，以达到打散 Region 热点的效果。详情可参考官网 [TiDB 专用系统变量和语法](/tidb-specific-system-variables.md#shard_row_id_bits)中 `SHARD_ROW_ID_BITS` 的介绍。
+TiDB 中以 Region 分片来管理数据库，通常来讲，TiDB 的热点指的是 Region 的读写访问热点。而 TiDB 中对于 PK 非整数或没有 PK 的表，可以通过设置 `SHARD_ROW_ID_BITS` 来适度分解 Region 分片，以达到打散 Region 热点的效果。详情可参考官网 [SHARD_ROW_ID_BITS](/shard-row-id-bits.md)中的介绍。
```
Comment (yikeke) [inline faq/sql-faq.md:217]:
```suggestion
TiDB 中以 Region 分片来管理数据库，通常来讲，TiDB 的热点指的是 Region 的读写访问热点。而 TiDB 中对于主键非整数或没有主键的表，可以通过设置 `SHARD_ROW_ID_BITS` 来适度分解 Region 分片，以达到打散 Region 热点的效果。详情可参考官网 [SHARD_ROW_ID_BITS](/shard-row-id-bits.md)中的介绍。

```

File: faq/sql-faq.md [hunk 8]
```diff
@@ -214,7 +214,8 @@ ID 没什么规律，只要是唯一就行，不过生成的时候，是有一
 
 ### 如何打散热点
 
-TiDB 中以 Region 分片来管理数据库，通常来讲，TiDB 的热点指的是 Region 的读写访问热点。而 TiDB 中对于 PK 非整数或没有 PK 的表，可以通过设置 `SHARD_ROW_ID_BITS` 来适度分解 Region 分片，以达到打散 Region 热点的效果。详情可参考官网 [TiDB 专用系统变量和语法](/tidb-specific-system-variables.md#shard_row_id_bits)中 `SHARD_ROW_ID_BITS` 的介绍。
+TiDB 中以 Region 分片来管理数据库，通常来讲，TiDB 的热点指的是 Region 的读写访问热点。而 TiDB 中对于非整数主键或没有主键的表，可以通过设置 `SHARD_ROW_ID_BITS` 来适度分解 Region 分片，以达到打散 Region 热点的效果。详情可参考官网 [SHARD_ROW_ID_BITS](/shard-row-id-bits.md)中的介绍。
+
```
Comment (yikeke) [inline faq/sql-faq.md:218]:
```suggestion

```

**Pull request merged**